### DISCUSSION TITLE
Welcome to EGSnrc Discussions!

### DISCUSSION THREAD

**Discussion by ftessier** [nrc-cnrc/EGSnrc#652] [General]
## 👋  Welcome!
  We’re using Discussions as a place to connect with members of the EGSnrc community. This forum is meant for asking questions you’re wondering about (without necessarily opening an issue), suggesting improvements to EGSnrc, sharing new ideas for development, etc. Note also that discussions and community user support are ongoing on the [EGSnrc sub-reddit](https://www.reddit.com/r/EGSnrc). Feel free to comment below in this thread with an introduction of yourself and what you do with EGSnrc, or jump right in and launch a new discussion thread!

**Comment by jianmingm** [comment]
Hi,

Recently I am comparing the EGSnrc results (egs_chamber) with the Geant4 results but failed to generate consistent results. I then went to see the physics reference of EGSnrc and Geant4 and found something confusing. 

The EGSnrc energy loss evaluation states that if the electron range is less than set values, the simulation of the current electron is terminated and its entire energy deposited locally. I am wondering whether "the entire energy" here includes the rest energy of electrons. The original text is listed below. 

(If the parameter i do rr is set to unity for the current region, the electron energy is less than e max rr (this is total energy, including rest energy!) and the electron range R is less than t⊥ (see section 2.4.10), the simulation of the current electron is terminated and its entire energy deposited locally.)

Then in the Geant4 Run Time Energy Loss Computation, if the energy of an electron is less than the _lowestEnergy_, the particle is stopped and the remaining kinetic energy is assigned to the local energy deposit. So only the kinetic energy will be deposited. 

If the EGSnrc codes are depositing the total energy of an electron, could this be why I am not getting consistent results between the two codes (my geometry is at the nanoscale)?

Thank you for the help!!
Best

**Reply by ftessier** [reply]
Indeed, EGSnrc only deposits the remaining kinetic energy of the electrons once they fall below ECUT, the cutoff energy for electron transport. Note that the value of ECUT is specified as the total energy. So if `ECUT = 521 keV` (i.e. 10 keV kinetic energy), electrons reaching this energy will be terminated and contribute 10 keV to the local energy deposition tally.

**Reply by jianmingm** [reply]
Thank you for the clarification!

**Comment by walleludvig** [comment]
Hi @ftessier,

I am writing here to discuss an issue I am having when attempting to run parallel jobs using the script supplied here: [https://egsnrcarchive.home.blog/2014/04/23/running-egsnrc-codes-on-multiple-cores-without-a-queuing-system/comment-page-1/#comment-5246](url).

Unfortunately Wordpress doesn't seem to allow me to post any comments on that discussion board, and so hope its okay that I decided to get in-touch here.

Thank you so much for providing the parallel jobs script.

I am trying to use it to run 10 parallel jobs but I get an abort trap 6 from each job:
”
line 154: <pid #> Abort trap: 6 $runcommand > /dev/null 2>&1
”

Any tips on how I might resolve this issue?

(I am running on Mac (M1 chip) (Monterey 12.5))

**Reply by rtownson** [reply]
Hi @walleludvig, welcome to github - that WordPress site is just an archive of our old google+ support page, which is why it is frozen. For new questions please post it as a new discussion instead of as a reply here.

But since I'm already answering, try this script:
https://gist.github.com/ftessier/7099aa0c6f9f7629d6e3

or read the script and see that it's actually very simple. You can launch each job on the command line one at a time like this:
```
egs_app -b -P 10 -j 1 >/dev/null 2>&1 &
egs_app -b -P 10 -j 2 >/dev/null 2>&1 &
etc...
```
the `>/dev/null 2>&1` part is just saving you from a bunch of output to the terminal, and the `&` is launching it as a background process. Change the `-j 1` to 2, 3, 4, etc for each job up to 10.
